core/src/impl/Kokkos_Atomic_Generic.hpp

```diff
@@ -287,11 +307,13 @@ KOKKOS_INLINE_FUNCTION T atomic_fetch_oper(
 }
 
 template <class Oper, typename T>
-KOKKOS_INLINE_FUNCTION T atomic_oper_fetch(
+KOKKOS_IMPL_THREAD_SANITIZER_IGNORE KOKKOS_INLINE_FUNCTION T atomic_oper_fetch(
     const Oper& op, volatile T* const dest,
     typename Kokkos::Impl::enable_if<(sizeof(T) != 4) && (sizeof(T) != 8)
-#if defined(KOKKOS_ENABLE_ASM) && \
-    defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST)
+#if defined(KOKKOS_ENABLE_ASM) &&                                              \
+    defined(KOKKOS_ACTIVE_EXECUTION_MEMORY_SPACE_HOST) &&                      \
+    0 /* TODO fix the preprocessor conditionals for the 128-bit CAS version of \
```

Comment:
TODONE?

Reply:
No, but there's a bug that this is working around. I don't remember why it's here because this pull request has been sitting here so long. The intention was to split this off into a separate issue, but I don't remember what that issue was